console/src/lib/tasks/NumberInput.svelte

```diff
@@ -24,7 +26,9 @@
 		fieldValidators.push(required);
 	}
 
-	let fieldValue = param.default;
+	const valueFromUrl = $page.url.searchParams.get(param.name);
+	let fieldValue = valueFromUrl ? Number(valueFromUrl) : param.default;
```

Comment:
I think you are casting fieldValue to a Number twice here. Also, if valueFromUrl is not a number, namely "12345a", then the result will be `NaN`. Have you checked that case? (I'm sure it's fine if it errors, but just that it is a sensible error.)

Reply:
@lukaso I've added the previous code back with comments. Let me know if it makes more sense.